iroh/src/client/blobs/batch.rs

```diff
@@ -0,0 +1,460 @@
+        &self,
+        reader: impl AsyncRead + Unpin + Send + 'static,
+        opts: AddReaderOpts,
+    ) -> anyhow::Result<TempTag> {
+        let AddReaderOpts { format, chunk_size } = opts;
+        let input = ReaderStream::with_capacity(reader, chunk_size);
+        self.add_stream_with_opts(input, format).await
+    }
+
+    /// Write a blob by passing bytes.
+    pub async fn add_bytes_with_opts(
+        &self,
+        bytes: impl Into<Bytes>,
+        format: BlobFormat,
+    ) -> Result<TempTag> {
+        let input = futures_lite::stream::once(Ok(bytes.into()));
+        self.add_stream_with_opts(input, format).await
+    }
+
+    /// Import a blob from a filesystem path.
+    ///
+    /// `path` should be an absolute path valid for the file system on which
+    /// the node runs, which refers to a file.
+    ///
+    /// If you use [ImportMode::TryReference], Iroh will assume that the data will not
+    /// change and will share it in place without copying to the Iroh data directory
+    /// if appropriate. However, for tiny files, Iroh will copy the data.
+    ///
+    /// If you use [ImportMode::Copy], Iroh will always copy the data.
+    ///
+    /// Will return a temp tag for the added blob, as well as the size of the file.
+    pub async fn add_file_with_opts(
+        &self,
+        path: PathBuf,
+        opts: AddFileOpts,
+    ) -> Result<(TempTag, u64)> {
+        let AddFileOpts {
+            import_mode,
+            format,
+        } = opts;
+        anyhow::ensure!(
+            path.is_absolute(),
+            "Path must be absolute, but got: {:?}",
+            path
+        );
+        anyhow::ensure!(path.is_file(), "Path does not refer to a file: {:?}", path);
+        let mut stream = self
+            .0
+            .rpc
+            .server_streaming(BatchAddPathRequest {
+                path,
+                import_mode,
+                format,
+                batch: self.0.batch,
+            })
+            .await?;
+        let mut res_hash = None;
+        let mut res_size = None;
+        while let Some(item) = stream.next().await {
+            match item?.0 {
+                BatchAddPathProgress::Abort(cause) => {
+                    Err(cause)?;
+                }
+                BatchAddPathProgress::Done { hash } => {
+                    res_hash = Some(hash);
+                }
+                BatchAddPathProgress::Found { size } => {
+                    res_size = Some(size);
+                }
+                _ => {}
+            }
+        }
+        let hash = res_hash.context("Missing hash")?;
+        let size = res_size.context("Missing size")?;
+        Ok((
+            self.local_temp_tag(HashAndFormat { hash, format }, Some(size)),
+            size,
+        ))
+    }
+
+    /// Add a directory as a hashseq in iroh collection format
+    ///
+    /// This can also be used to add a single file as a collection, if
+    /// wrap is set to [WrapOption::Wrap].
```

Comment:
```suggestion
    /// Adds a directory as a hashseq in iroh collection format.
    ///
    /// This can also be used to add a single file as a collection, if
    /// wrap is set to [`WrapOption::Wrap`].
```